core/src/main/java/org/ethereum/beacon/core/operations/Attestation.java

```diff
@@ -79,6 +81,24 @@ public BLSSignature getSignature() {
     return signature;
   }
 
+  public Attestation withAggregationBits(Bitlist aggregationBits, SpecConstants specConstants) {
+    return new Attestation(
+        ensureMaxSize(aggregationBits, specConstants), data, custodyBits, BLSSignature.ZERO);
+  }
+
+  public Attestation withData(AttestationData data) {
+    return new Attestation(aggregationBits, data, custodyBits, BLSSignature.ZERO);
```

Comment:
What if we add an explicit check like `assert signature == BLSSignagure.ZERO`?

Reply:
To forbid updating already signed attestations?

Reply:
Signed attestation looks like an invariant that we might want to preserve. WDYT?